### ISSUE TITLE
Mutex requires data to by Sync

### ISSUE THREAD

**Issue by peopleskai** [lobaro/FreeRTOS-rust#74]
Hi,

In line https://github.com/lobaro/FreeRTOS-rust/blob/master/freertos-rust/src/mutex.rs#L11 we can see the `Mutex` API requires the data it holds to have `Sync` trait for it to implement `Sync`. 

I wanted to understand the reason for this. I believe in the standard library `Mutex` will be `Send+Sync` if the data type is `Send`. This makes sense since `Mutex` is what provides the `Sync` feature for the data. If the data is already `Sync`, why would it need to be wrapped in a `Mutex`?

**Comment by schteve** [comment]
I think your suspicions are correct, what you say makes sense to me. The mutex should only need the data to be Send to implement both Send and Sync.

**Comment by peopleskai** [comment]
Yes i think it makes sense to me, and this change should be backwards compatible since it would reduce the requirements on the data for mutex to be sync

**Comment by peopleskai** [comment]
Do we have maintainers with bandwidth to make this change? Or should I try to make the change and get you to review it?

**Comment by schteve** [comment]
Feel free to make the change if you like. I'm assuming it's just removing Sync from those two lines, but I don't know if there's any additional changes needed.

Good point about the backward compatibility - always nice to have that.

**Comment by michael-p** [comment]
I assume the exact same argument also holds for `ExclusiveData`, and hence the trait bounds for that type should also only require `T: Sync`? If others agree I can send in a PR.
